infrastructure/cloudconnect/README.md

```diff
@@ -0,0 +1,205 @@
+
+![](./pics/tunnel_established.png "Tunnel Established")
+
+![](./pics/dropped_sent_packets.png "Dropped Sent Packets")
+
+![](./pics/dropped_received_packets.png "Dropped Received Packets")
+
+![](./pics/conditions.png "Conditions")
+
+
+## Testing
+
+One can query the AWS VPN Gateway to provide status of the tunnels. In both cases, status should be _UP_, with the BGP connection reading _XX BGP ROUTER_ (meaning BGP has propagated the routes properly)
+
+    ## Static Route
+    $ aws ec2 describe-vpn-connections --filter Name=vpn-connection-id,Values=vpn-914251f0 --query VpnConnections[0].VgwTelemetry
+    [
+        {
+            "Status": "UP",
+            "AcceptedRouteCount": 1,
+            "OutsideIpAddress": "52.4.54.162",
+            "LastStatusChange": "2017-03-14T03:45:03.000Z",
+            "StatusMessage": ""
+        },
+        {
+            "Status": "UP",
+            "AcceptedRouteCount": 1,
+            "OutsideIpAddress": "54.173.28.19",
+            "LastStatusChange": "2017-03-14T03:45:44.000Z",
+            "StatusMessage": ""
+        }
+    ]
+
+    ## BGP
+    $ aws ec2 describe-vpn-connections --filter Name=vpn-connection-id,Values=vpn-8e4251ef --query VpnConnections[0].VgwTelemetry
+    [
+        {
+            "Status": "UP",
+            "AcceptedRouteCount": 22,
+            "OutsideIpAddress": "34.193.213.250",
+            "LastStatusChange": "2017-03-14T03:53:39.000Z",
+            "StatusMessage": "22 BGP ROUTES"
+        },
+        {
+            "Status": "UP",
+            "AcceptedRouteCount": 22,
+            "OutsideIpAddress": "34.193.235.205",
+            "LastStatusChange": "2017-03-14T03:55:10.000Z",
+            "StatusMessage": "22 BGP ROUTES"
+        }
+    ]
+
+### Iperf
+
+Performance with Iperf: In this example, 3 instances are launched in GCP and 3 in AWS. Four of the instances (2 in each environment) should live in subnets supported by the static route connection, and the last pair is intended to test the BGP connection.  
+
+#### GCP -> AWS
+
+  **NOTE:** Try running all client commands at the same time
+
+- On each AWS instance, run `iperf3 -i1 -s`
+- On each GCP instance, run `iperf3 -P2 -c <aws_internal_ip> -t300`
+- Upon completion, reference StackDriver Sent Bytes and Dropped Packets
+
+![](./pics/send_test.png "Send Test")
+
+#### AWS -> GCP
+
+- On each GCP instance, run `iperf3 -i1 -s`
+- On each AWS instance, run `iperf3 -P2 -c <gcp_internal_ip> -t300`
+- Upon completion, reference StackDriver Received Bytes and Dropped Packets
+
+![](./pics/receive_test.png "Receive Test")
+
+#### Failover Testing
+
+This test will be run on two nodes within the static route supported subnets.
+
+- On the one of the AWS, ping a corresponding GCP instance
+- Using gcloud, kill the static vpn deployments `gcloud deployment-manager deployments delete vpn-static-to-aws`
+- Verify that one gets an alert (if configured above), and ping reestablishes automatically within 1 minute.
```

Comment:
Extra credit for adding the commands for cleaning up:  

gcloud deployment-manager  deployments delete ...
aws cloudformation delete-stack --stack-name gcp-vpn-bgp

etc..